### PR TITLE
fw-4201, sites API 

### PR DESCRIPTION
**Pull request by sarahfirstvoices**
In this review:
- site list API (public list of sites, grouped by language), with tests
- site detail API (basic details about a site, for authorized users), with tests
- custom pagination class optimized for React's useInfiniteQuery (ended up not being used for site list, but it's enabled by default for future APIs)
- API docs available at /api/docs
- Browsable API re-enabled for local, to allow easier db profiling

NOT in this review:
- list of a user's site memberships (fw-4257)
- applying permissions to list views without extra db queries (fw-4255)
- standardized error responses (fw-4256)

### CODE REVIEW

File: src/config/settings/base.py [hunk 1]
```diff
@@ -256,6 +255,13 @@
         }
     },
     "root": {"level": "INFO", "handlers": ["console"]},
+    # "loggers": {
+    #     "rules": {
+    #         "handlers": ["console"],
+    #         "level": "DEBUG",
+    #         "propagate": True,
+    #     },
+    # },
```
Comment (sarahfirstvoices) [inline src/config/settings/base.py:264]: I can remove this if folks prefer-- it's the logging config for seeing which permissions are being evaluated

File: src/firstvoices/backend/fixtures/appjson-defaults.json [hunk 2]
```diff
@@ -0,0 +1 @@
...: "about", "title": "About", "transKey": "general.about", "itemsData": [{"href": "/our-language", "title": "Our Language", "transKey": "general.our_language"}, {"href": "/our-people", "title": "Our People", "transKey": "general.our_people"}]}, "learn": {"id": "learn", "title": "Learn", "transKey": "general.learn", "itemsData": [{"href": "/songs", "title": "Songs", "transKey": "general.songs"}, {"href": "/stories", "title": "Stories", "transKey": "general.stories"}, {"href": "/games", "title": "Games", "transKey": "general.games"}]}, "resources": {"id": "resources", "title": "Resources", "transKey": "general.resources", "itemsData": [{"href": "/apps", "title": "Mobile App", "transKey": "general.apps"}, {"href": "/keyboards", "title": "Keyboards", "transKey": "general.keyboards"}]}, "dictionary": {"id": "dictionary", "title": "Dictionary", "transKey": "general.dictionary", "itemsData": [{"href": "/words", "title": "Words", "transKey": "general.words"}, {"href": "/phrases", "title": "Phrases", "transKey": "general.phrases"}, {"href": "/alphabet", "title": "Alphabet", "transKey": "general.alphabet"}, {"href": "/categories", "title": "Categories", "transKey": "general.categories"}]}}}}]
```
Comment (sarahfirstvoices) [inline src/firstvoices/backend/fixtures/appjson-defaults.json:1]: @sahil97 I know you looked into pros and cons of including the pk in fixtures. Can you remind me what the conclusion was?

File: src/firstvoices/backend/tests/test_predicates/test_predicate_utils.py [hunk 3]
```diff
@@ -52,4 +52,4 @@ def test_get_app_role_for_non_member(self):
 
     def test_get_app_role_for_member(self):
         member_user = get_app_admin(AppRole.STAFF)
-        assert utils.get_site_role(member_user) == AppRole.STAFF
+        assert utils.get_app_role(member_user) == AppRole.STAFF
```
Comment (sarahfirstvoices) [inline src/firstvoices/backend/tests/test_predicates/test_predicate_utils.py:55]: unrelated test fix :)

File: src/firstvoices/backend/views/base_views.py [hunk 4]
```diff
@@ -1,17 +1,31 @@
...
 from rules.contrib.rest_framework import AutoPermissionViewSetMixin
 
+from firstvoices.backend.predicates import utils
+
 
 class FVPermissionViewSetMixin(AutoPermissionViewSetMixin):
     """
-    Mixin used to override the default list view and enable the use of rules based permissions as defined in a model.
-    If this mixin is used, the list view will return all model instances that the user has view permissions on.
+    Enforces object-level permissions in ``rest_framework.viewsets.ViewSet``,
```
Comment (sarahfirstvoices) [inline src/firstvoices/backend/views/base_views.py:9]: I updated this to include the relevant info from the super class docstring

File: src/config/settings/local.py [hunk 5]
```diff
@@ -22,13 +22,18 @@
...
+        "rest_framework.renderers.BrowsableAPIRenderer",
     ),
+    # these authentication classes are for local only
     "DEFAULT_AUTHENTICATION_CLASSES": (
         "rest_framework.authentication.SessionAuthentication",
         "rest_framework.authentication.TokenAuthentication",
     ),
     "DEFAULT_PERMISSION_CLASSES": ("rest_framework.permissions.IsAuthenticated",),
     "UNAUTHENTICATED_USER": None,
+    "DEFAULT_SCHEMA_CLASS": "drf_spectacular.openapi.AutoSchema",
```
Comment (sahil97) [inline src/config/settings/local.py:34]: Should we add it to both places ? Or we can just make the second object extend from the first one, i.e. local can add on to the base REST_FRAMEWORK dict, the overriding of the variable can make it confusing.

Comment (sarahfirstvoices) [reply src/config/settings/local.py]: Ohhh yes, it would absolutely be better that way

File: src/firstvoices/backend/fixtures/appjson-defaults.json (hunk 2 above)
Comment (sarahfirstvoices) [reply src/firstvoices/backend/fixtures/appjson-defaults.json:1]: Thanks, I'll leave it for this one then to prevent duplicates.

**Review by sahil97 — APPROVED** [review]
